src/Microsoft.Health.Fhir.Core/Features/Search/Parameters/SearchParameterOperations.cs

```diff
@@ -107,6 +107,11 @@ public async Task DeleteSearchParameterAsync(RawResource searchParamResource)
         {
             try
             {
+                // We need to make sure we have the latest search parameters before trying to delete
+                // existing search parameter. This is to avoid trying to update a search parameter that
+                // was recently added and that hasn't propogated to all fhir-server instances.
+                await GetAndApplySearchParameterUpdates(CancellationToken.None);
```

Comment:
I noticed that we also call this in `UpdateSearchParameterAsync()`. 

For my understanding, why isn't this needed in `AddSearchParameterAsync()` as well? I'm wondering if there a scenario where we could get duplicate search parameters if a newly added param isn't synced to all instances

Reply:
Not an author, but I spend some time thinking about it, and I would say, what delete and update prefer to have updated collection since you modify it in a certain way and you want to make sure you modify latest version.

With adding new search parameter only thing you care about is uniquness of search parameter, and I think it's get validated before we get to this place.